pkg/kubelet/config/http_test.go

```diff
@@ -443,17 +411,17 @@ func TestExtractPodsFromHTTP(t *testing.T) {
 					},
 				}),
 		},
-		{
-			desc:     "Empty Array",
-			pods:     []v1beta3.Pod{},
-			expected: CreatePodUpdate(kubelet.SET, kubelet.HTTPSource),
-		},
 	}
 
 	for _, testCase := range testCases {
-		data, err := json.Marshal(testCase.pods)
+		var versionedPods runtime.Object
+		err := testapi.Converter().Convert(&testCase.pods, &versionedPods)
```

Comment:
If we want fully valid objects, we might need to convert back to the internal rep.